core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -475,6 +478,13 @@ public void reconfigure(ServerConfigurator serverConfigurator) {
         // Invoke the serviceAdded() method in Service so that it can keep the reference to this Server or
         // add a listener to it.
         config.serviceConfigs().forEach(cfg -> ServiceCallbackInvoker.invokeServiceAdded(cfg, cfg.service()));
+        hasWebSocketService = hasWebSocketService(config);
+    }
+
+    private static boolean hasWebSocketService(UpdatableServerConfig config) {
+        return config.serviceConfigs()
+                     .stream()
+                     .anyMatch(serviceConfig -> serviceConfig.service().as(WebSocketService.class) != null);
```

Comment:
Note: If a server doesn't have a websocket service but later adds a websocket service via `Server#reconfigure`, users still won't be able to use websocket with http2

Reply:
> users still won't be able to use websocket with http2

To be precise, they can't for the existing connections but can for the new ones, right? I would say it's a bug otherwise.

Reply:
> This method is also called from Server.reconfigure(...) 😉

Sure, but when actually using the field we don't use a field from `ServerConfig` but we pass the parameter in directly don't we? Anyways, no problem if this case is handled correctly

https://github.com/line/armeria/blob/c05558ed67e30b4922bd5fa58d90829326c423a8/core/src/main/java/com/linecorp/armeria/server/Server.java#L565